### PR TITLE
Add support for client tags in Python API

### PR DESCRIPTION
**Pull request by mdesmet**
Fixes #154 

Co-authored-by: abrisan <s1505883@sms.ed.ac.uk>

### CODE REVIEW

File: tests/integration/test_dbapi_integration.py [hunk 1]
```diff
@@ -27,7 +28,7 @@ def trino_connection(run_trino):
     _, host, port = run_trino
 
     yield trino.dbapi.Connection(
-        host=host, port=port, user="test", source="test", max_attempts=1
+        host=host, port=port, user="test", source="test", max_attempts=1, client_tags=["integration_test"]
```
Comment (ebyhr) [inline tests/integration/test_dbapi_integration.py:31]: Please create a connection with client tags in `test_client_tags`, not here.

File: tests/integration/test_dbapi_integration.py [hunk 2]
```diff
@@ -859,3 +860,17 @@ def test_info_uri(trino_connection):
...
+    cur.execute('SELECT 1')
+    cur.fetchall()
+
+    api_url = "http://" + trino_connection.host + ":" + str(trino_connection.port)
+    query_info = requests.post(api_url + "/ui/login", data={
+        "username": "admin",
+        "password": "",
+        "redirectPath": api_url + '/ui/api/query/' + cur._query.query_id
+    }).json()
+    assert query_info['session']['clientTags'] == ["integration_test"]
```
Comment (ebyhr) [inline tests/integration/test_dbapi_integration.py:876]: Please add more test cases:
* multiple tags. e.g. `tag1,tag2`
*  tag value has `=`

File: tests/integration/test_dbapi_integration.py [hunk 3]
```diff
@@ -859,3 +860,78 @@ def test_info_uri(trino_connection):
...
+def test_client_tags_single_tag(run_trino):
+    _, host, port = run_trino
+
+    client_tags = ["foo"]
+    trino_connection = trino.dbapi.Connection(
+        host=host,
+        port=port,
+        user="test",
+        source="test",
+        max_attempts=1,
```
Comment (ebyhr) [inline tests/integration/test_dbapi_integration.py:874]: Remove unnecessary arguments, `source` and `max_attempts` unrelate to client tag test.

File: tests/integration/test_dbapi_integration.py [hunk 4]
```diff
@@ -859,3 +860,78 @@ def test_info_uri(trino_connection):
...
+    cur = trino_connection.cursor()
+    cur.execute('SELECT 1')
+    cur.fetchall()
+
+    api_url = "http://" + trino_connection.host + ":" + str(trino_connection.port)
+    query_info = requests.post(api_url + "/ui/login", data={
+        "username": "admin",
+        "password": "",
+        "redirectPath": api_url + '/ui/api/query/' + cur._query.query_id
+    }).json()
```
Comment (ebyhr) [inline tests/integration/test_dbapi_integration.py:936]: Could you add a helper method that takes a `client_tags` argument and returns `query_info['session']['clientTags']`?

**Review by ebyhr — APPROVED** [review]

File: tests/integration/test_dbapi_integration.py (hunk 3 above)
Comment (ebyhr) [reply tests/integration/test_dbapi_integration.py:874]: Reminder.

File: tests/integration/test_dbapi_integration.py (hunk 4 above)
Comment (ebyhr) [reply tests/integration/test_dbapi_integration.py:932]: Let's move `trino.dbapi.Connection` to this method.

File: tests/integration/test_dbapi_integration.py (hunk 3 above)
Comment (mdesmet) [reply tests/integration/test_dbapi_integration.py]: Missed that one!

**Pull request merged**